### PR TITLE
Add HTML block with preview.

### LINKED ISSUE #1386: Add Block: Custom HTML
Let's build a block that allows you to paste custom HTML into it.

This mockup includes a new pattern: Text button groups. It's exactly the same as the control group containing alignments, in that you can only pick one button at a time in the group. But this one has text instead of icons inside. 

## Attributes

- Code & Preview tabs

Clicking the preview tab shows a preview of the form instead of the HTML. 

Syntax highlighting and line numbers is a "nice to have", and doesn't need to be in this block in V1.

## States

Neutral:

custom html

Selected:

custom html selected

### PR DESCRIPTION
**Pull request by mtias**
image

Closes #1386.

### CODE REVIEW

File: blocks/library/html/index.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+		render() {
+			const { preview } = this.state;
+			const { attributes, setAttributes, focus } = this.props;
+
+			return (
+				<div>
+					{ focus &&
+						<BlockControls key="controls">
+							<ul className="components-toolbar">
```
Comment (youknowriad) [inline blocks/library/html/index.js]: Can't we use the regular `Toolbar` component for this? I guess we just need to make the icon optional and drop the fixed width?

Comment (mtias) [inline blocks/library/html/index.js]: It forces icon at the moment. And we'll probably just create a segmented-control out of this in any case.

**Comment by jasmussen** [comment]
Designed new icon for the block here: https://github.com/WordPress/dashicons/pull/207

File: blocks/library/html/index.js (hunk 1 above)
Comment (youknowriad) [inline blocks/library/html/index.js]: as part of #1256 I updated the `Toolbar` component to allow this

Comment (youknowriad) [inline blocks/library/html/index.js]: Could be done separately of course

**Review by youknowriad — APPROVED** [review]
Works great, feel free to merge when the tests are fixed

**Pull request merged**